plugins/ecs-serviceconnect/plugin/commands.go

```diff
@@ -0,0 +1,244 @@
+	mode config.NetworkMode,
+	cidr string,
+	redirectPort string) error {
+	var err error
+	switch mode {
+	case config.AWSVPC, config.BRIDGE_DNAT:
+		err = iptable.Append("nat", "OUTPUT", "-p", "tcp", "-d", cidr,
+			"-j", "REDIRECT", "--to-port", redirectPort)
+	case config.BRIDGE_TPROXY:
+		// Create a new chain
+		if err = iptable.NewChain("mangle", egressChain); err != nil {
+			return err
+		}
+		//TODO: Add other rules
+	}
+	if err != nil {
+		log.Errorf("Append rule to redirect traffic of CIDR failed: %v", err)
+	}
+	return err
+}
+
+// Del is the internal implementation of CNI DEL command.
+// CNI DEL command can be called by the orchestrator multiple times for the same interface,
+// and thus must be best-effort and idempotent.
+func (plugin *Plugin) Del(args *cniSkel.CmdArgs) error {
+	// Parse network configuration.
+	netConfig, err := config.New(args)
+	if err != nil {
+		log.Errorf("Failed to parse netconfig from args: %v.", err)
+		return err
+	}
+
+	log.Infof("Executing DEL with netconfig: %+v.", netConfig)
+
+	// Search for the target network namespace.
+	ns, err := netns.GetNetNS(args.Netns)
+	if err != nil {
+		log.Errorf("Failed to find netns %s: %v.", args.Netns, err)
+		return err
+	}
+
+	// Delete IP rules in the target network namespace.
+	err = ns.Run(func() error {
+		for _, proto := range netConfig.IPProtocols {
+			if err = plugin.deleteIptablesRules(proto, netConfig); err != nil {
+				log.Errorf("Failed to delete ip rules: %v.", err)
+				return err
+			}
+		}
+		return nil
+	})
+	return err
+}
+
+// deleteIptablesRules removes iptables/ip6tables rules in container network namespace.
+func (plugin *Plugin) deleteIptablesRules(
+	proto iptables.Protocol,
+	config *config.NetConfig) error {
+	/// Create a new iptables session.
+	iptable, err := iptables.NewWithProtocol(proto)
+	if err != nil {
+		return err
+	}
+
+	// Delete Ingress rules
+	if err = plugin.deleteIngressRules(iptable, config); err != nil {
+		return err
+	}
+
+	// Delete Egress rules
+	return plugin.deleteEgressRules(iptable, proto, config)
+}
+
+// deleteIngressRules deletes the iptable rules for ingress traffic.
+func (plugin *Plugin) deleteIngressRules(
+	iptable *iptables.IPTables,
+	config *config.NetConfig) error {
+	// Skip deleting rules when there is no redirection
+	if len(config.IngressRedirectToListenerPortMap) == 0 {
+		return nil
+	}
+	// Delete ingress rule from iptables.
+	if err := pluginutils.DeleteNonLocalRedirectionRules(iptable, ingressChain); err != nil {
+		log.Errorf("Delete the rule in PREROUTING chain failed: %v", err)
+		return err
+	}
+	// Remove the added chain
+	return pluginutils.RemoveChain(iptable, ingressChain)
+}
+
+// deleteEgressRules deletes the iptable rules for egress traffic.
+func (plugin *Plugin) deleteEgressRules(
+	iptable *iptables.IPTables,
+	proto iptables.Protocol,
+	config *config.NetConfig) error {
+	// Skip deleting rules when there is no egress listener port
+	if config.EgressPort == 0 {
```

Comment:
although this will/should never really happen right?

Reply:
No. When there are no other services in the namespace, the egress port will not be present.